dali/pipeline/operator/builtin/external_source.cc

```diff
@@ -92,6 +92,11 @@ The memory location must match the specified ``device`` parameter of the operato
 For the CPU, the provided memory can be one contiguous buffer or a list of contiguous Tensors.
 For the GPU, to avoid extra copy, the provided buffer must be contiguous. If you provide a list
 of separate Tensors, there will be an additional copy made internally, consuming both memory
-and bandwidth.)code", false);
+and bandwidth.)code", false)
+  .AddOptionalArg("dtype", R"code(Input data type.
```

Comment:
I guess it's a topic for further discussion, but I think we should announce that this parameter is going to be mandatory (and possibly start issuing warnings if it isn't provided?).

Reply:
I added information to the docstring. 
The warning should go to a separate PR that will also add the argument to all of our internal uses of external source.